### ISSUE TITLE
Unified Alerting - Data sources selected, but alert rules from all sources are constantly re-loaded

### ISSUE THREAD

**Issue by gelicia** [grafana/grafana#82542]
### Discussed in https://github.com/grafana/grafana/discussions/72913

<div type='discussions-op-text'>

<sup>Originally posted by **mac133k** August  4, 2023</sup>
**Current behaviour:**
This is a very useful feature to allow selecting one or more datasources in Alerting > Alert rules (https://github.com/grafana/grafana/issues/62969), but it seems like even if only one or more datasources are selected and the view is filtered to only display the rules from those, all datasources are being checked and a message is displayed saying 'Loading rules from <number> datasources".

**Expected behavriour:**
Only rules from selected datasource(s) are being re-loaded in Alert rules GUI.

Grafana version: 10.0.1</div>

**Comment by PrajwalParvati** [comment]
Hello, @gelicia , is this open for contribution? I'd like to work on this feature.

**Comment by gelicia** [comment]
> Hello, [@gelicia](https://github.com/gelicia) , is this open for contribution? I'd like to work on this feature.

@PrajwalParvati Yep, this seems to be in the alerting backlog but not prioritized, so it is open to contribution as far as I'm aware.

**Comment by gelicia** [comment]
@Rkejji I have not heard any new info since last October, so feel free.
